File: Gateway/app/src/main/java/net/mfuertes/aagw/gateway/BootCompletedReceiver.java
package net.mfuertes.aagw.gateway;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.hardware.usb.UsbManager;

public class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {

        /*
        TBR!!!
             if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
                val intentFilter = IntentFilter()
                intentFilter.addAction(WifiManager.NETWORK_STATE_CHANGED_ACTION)
                context.applicationContext.registerReceiver(this, intentFilter)
                toggleUSB()
            } else if (WifiManager.NETWORK_STATE_CHANGED_ACTION == intent.action) {
                if (isNetworkAvailable(context)) {
                    toggleUSB()
                }
            }
         */
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            UsbHelper.setMode(context.getSystemService(UsbManager.class), UsbHelper.FUNCTION_MTP);
        }
    }
}

File: Helper/app/src/main/java/net/mfuertes/aagw/helper/BootCompletedReceiver.kt
package net.mfuertes.aagw.helper

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.hardware.usb.UsbManager
class BootCompletedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        PairingReceiver.makeDiscoverable()
        UsbHelper.setMode(context.getSystemService(UsbManager::class.java) as UsbManager, UsbHelper.FUNCTION_MTP)
    }
}